### PR TITLE
Add Qdrant as a long memory backend.

### PR DESCRIPTION
**Pull request by wangxuqi**
Hi there!
For this pull request, I have read your announcement carefully.

###  Background
As I notice there are many vector databases provided in [chatgpt-retrieve-plugin.](https://github.com/openai/chatgpt-retrieval-plugin/tree/main/datastore/providers)
I think all the vector databases can be used by you as a long-term memory. So this is the first commit to add the most easier used one 'Qdrant'. And I am willing to implement them all!

### Changes
This commit only  "Add Qdrant as a long memory backend."
what I do is:
- Implement QdrantMemory based on MemoryProviderSingleton
- add 'qdrant' option in "memory/__init__.py"
- add 'qdrant-client' in requirements.txt
- add unit test and integration test in 'qdrant_memory_test.py' and 'qdrant_memory_tests.py' 

### Documentation
The implement is very clear in-code comments. I just implement all your API and test all them to work.
- def add(self, data)
- def get(self, data)
- def clear(self)
- def get_relevant(self, data, num_relevant=5)
- def get_stats(self)

### Test Plan
- Unit test in 'qdrant_memory_test.py' 
- IntegrationTest in 'qdrant_memory_tests.py' 
- Test step: Lauch a Qdrant docker instance `docker run -p "6333:6333" -p "6334:6334" qdrant/qdrant:v1.0.3`  and then just run all the related cases.
- I've also tested with the following instructions:
```
Enter up to 5 goals for your AI:  For example: Increase net worth, Grow Twitter Account, Develop and manage multiple businesses autonomously'
Enter nothing to load defaults, enter nothing when finished.
Goal 1: Visit a website which provides today's news.
Goal 2: Summarize the articles and write the results to text files
Goal 3: Terminate
Goal 4: 
Using memory of type: QdrantMemory
SUMMARIZE TODAY'S LEADING NEWS THOUGHTS:  I need to determine which news website to visit and then summarize 
...
```

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Review by nponeccop — APPROVED** [review]

**Comment by nponeccop** [comment]
@wangxuqi There are conflicts now

**Comment by wangxuqi** [comment]
> @wangxuqi There are conflicts now

The conflicts is caused by another long memory  'Milvus' merged.  I have resolved the conflict.

**Comment by wangxuqi** [comment]
@nponeccop As I solve the conflict , I  find the merged 'Milvus memory' code is very poor. With no tests and even not worked. Also the package import is wrong and the milvs.py is not under the correct directory. 
Please review the pull request again https://github.com/Significant-Gravitas/Auto-GPT/pull/801 or I will fixed it later.

**Review by nponeccop — APPROVED** [review]

**Comment by wangxuqi** [comment]
Fix conflict with another merged PR [1658](https://github.com/Significant-Gravitas/Auto-GPT/pull/1658). 
As I see you have add a potential plugin label. I think it maybe a better way to fit all [chatgpt-retrieve-plugin.](https://github.com/openai/chatgpt-retrieval-plugin/tree/main/datastore/providers) vector databases. Try to do this later.

File: README.md
Comment (nponeccop) [inline README.md]: Don't remove milvus

**Review by nponeccop — CHANGES_REQUESTED** [review]

File: README.md
Comment (wangxuqi) [reply README.md]: Milvus Setup is duplicated. You can see line 367 to 377

**Review by nponeccop — APPROVED** [review]

**Comment by wangxuqi** [comment]
Ops! Conflict with `README.md` again.  resolve it. Hope this time no more conflicts.

**Comment by wangxuqi** [comment]
CI broken fixed by https://github.com/Significant-Gravitas/Auto-GPT/pull/2061

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by ntindle** [comment]
Hey, I've marked this as don't merge until the Memory Fixes are in. Sorry to keep it on hold longer, just no real way to test/validate functionality until the fixes are in

**Comment by wangxuqi** [comment]
> Hey, I've marked this as don't merge until the Memory Fixes are in. Sorry to keep it on hold longer, just no real way to test/validate functionality until the fixes are in

As I see the discussion https://github.com/Significant-Gravitas/Auto-GPT/discussions/4280. So close the pr for now. Thanks for your attention and reply.  Wish you all good.

**Pull request closed without merging**
